doc/default_route.md

```diff
@@ -0,0 +1,56 @@
+# Default Route
+
+Linkmgrd is expected to react on switch's default route changes. Orchagent will update both IPv4 and Ipv6 default route status to state db entries. If there is a valid default route, it shall be updated as `state:ok`. If there isn't a valid default route, it shall be updated as `state:na`. 
+
+For now we only care about IPv4, but state db notification handler should be properly defined for both entries. 
+
+## Requirements (active-standby)
+1. If one ToR is missing default route, it should be toggled to `standby`.
+1. If one ToR is missing default route, it should remain in `standby`. It shouldn't switch to `active` even if peer ToR appears to be unhealthy. 
+1. If both ToRs are missing default route, oscillation should be avoided. Mux direction should be parked on one side. 
+1. dualToR should be able to recover automatically if a valid default route is added back.
+1. If in manual mode, default route change shouldn't trigger mux direction change.  
+1. User should have the option to turn off this feature of reacting default route changes. 
+
+## Proposed Solution
+
+### Solution One (current implementation)
+* Stop sending heartbeat   
+To shutdown heartbeat sending if default route is missing, is a straightforward way to fake an unhealthy status. A faked unhealthy status will help linkmgrd:
+    * keep the ToR in standby if one ToR is missing default route;
+    * avoid oscillation if both ToRs are missing defualt route (as eventually both will be `LinkProber:Wait` state).   
+
+Link prober will still be able to receive heartbeat. So if we remove & re-add the default route on both ToRs, a state transition path can be:   
+
+```
+         0.init state   ->  1.remove on A   ->  2.remove on B     -> 3.re-add on B      -> 4.re-add on A
+ToR A   active, healthy ->  standby, healthy->  active, unhealthy ->  standby, healthy  -> standby, healthy
```

Comment:
Should 1 be standby, unhealthy?

Reply:
Why would step 2 toggle when they both lost default gw?

Reply:
It should be healthy. Though sending is suspended, it can still receive peer's HB.

Reply:
Step 3 should also be standby unhealthy

Reply:
> Why would step 2 toggle when they both lost default gw?

Addressed in section below ` Avoid proactive switches `.

Reply:
> Step 3 should also be standby unhealthy

Same as Step 1.

Reply:
Discussed offline, resolving conversation.